ex4/ex4.c

```diff
@@ -10,6 +10,18 @@
 int main(int argc, char* argv[])
 {
     // Your code here    
+    int rc = fork();
 
+    if (rc < 0) { // Fail -> exit
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+      } else if (rc == 0) { // Child
+        printf("Child\n");
+        execl("ls", NULL);
```

Comment:
My compiler yelled a bit about this line, though all warnings (compile still succeeded and program output looks correct). Specifically, this throws 'null argument where non-null required (argument 2) [-Wnonnull]` and `not enough variable arguments to fit a sentinel [-Wformat=]` - while warnings may not prevent compilation, they should always be acknowledged and addressed when possible in order to avoid unexpected behavior, I would encourage you all to look into exactly what these mean and why they're occurring. Good work here altogether!